consensus/polybft/consensus_runtime.go

```diff
@@ -477,6 +463,13 @@ func (c *consensusRuntime) restartEpoch(header *types.Header) error {
 	c.lock.Lock()
 	c.epoch = epoch
 	c.lastBuiltBlock = header
+
+	proposerSnapshot := c.proposerSnapshot.Copy()
+	if err := c.updateProposerSnapshot(proposerSnapshot, header.Number); err != nil {
```

Comment:
This will be changed with EVM-303. The problem is currently we have condition:
- if end of epoch -> call restart epoch and change state of runtime (lastblock, snapshot, epoch)
- otherwise change lastblock, snapshot